ash/src/entry.rs

```diff
@@ -226,3 +231,40 @@ impl<L> EntryCustom<L> {
         }
     }
 }
+
+impl vk::StaticFn {
+    pub fn load_checked<F>(mut _f: F) -> Result<Self, MissingEntryPoint>
+    where
+        F: FnMut(&::std::ffi::CStr) -> *const c_void,
+    {
+        // TODO: Make this a &'static CStr once CStr::from_bytes_with_nul_unchecked is const
+        static ENTRY_POINT: &[u8] = b"vkGetInstanceProcAddr\0";
+
+        Ok(vk::StaticFn {
+            get_instance_proc_addr: unsafe {
+                let cname = CStr::from_bytes_with_nul_unchecked(ENTRY_POINT);
+                let val = _f(cname);
+                if val.is_null() {
+                    return Err(MissingEntryPoint(&ENTRY_POINT[..ENTRY_POINT.len() - 1]));
+                } else {
+                    ::std::mem::transmute(val)
+                }
+            },
+        })
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct MissingEntryPoint(&'static [u8]); // Slice without the NUL terminator
```

Comment:
Including the bytes there is a bit odd. This will always be `vkGetInstanceProcAddr`. Any reason this is not an empty struct?

Reply:
This should probably be a more generic `MissingSymbol` if the name is kept.

Reply:
I've changed it to an empty struct.